_posts/2020-08-13-Goals_general_architecture.md

```diff
@@ -0,0 +1,27 @@
+---
+layout: post
+category: architecture
+title: Goals & general architecture
+---
+
+When building Auditree we wanted to shift compliance problems into engineering problems. We have more people with knowledge and expertise around engineering, DevOps and automation than we do around compliance, so scaling the compliance skills we do have is essential. At the same time; tools & approaches that are familiar to our engineers is going to mean better engagement from them in addressing & supporting compliance activities.
+
+![Clipboard to automation](/images/clipboard.png)
+
+When designing Auditree, we deliberately constrained the architecture to only use tools that were already in our stack, and had a goal of not requiring any additional infrastructure be added or managed by the team. This makes sense not just from a familiarity & velocity point of view, but also from a repudiation point of view; if we're using systems managed as a service by other teams we have less access & ability to tamper with them.
+
+## Framework is smart so fetchers & checks don't need to be
+
+The reason for building Auditree is to improve the scaling of our compliance efforts. We wanted engineers to be able to dip in, write a `unittest` based check & go back to their normal activities. To enable this, we built the framework, with the aim of it holding the "smarts" and the fetchers/checks only containing the logic and sufficient intelligence required to collect & evaluate evidence.
+
+We've exposed this functionality as [decorators and context managers][evidence], as well as [`Fetcher`][fetcher] and [`Check`][check] classes. Using these mean people can write their code and not worry about evidence TTL, where files reside in the locker or managing git as the framework takes care of these functions for you.
```

Comment:
I didn't want to link to line numbers, since they'll be wrong immediately after the next update. Good call onlinking to the design principles. I wanted them to be code e.g. a `Fetcher` not a fetcher.